src/Components/Server/src/JsonContext.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Components.Web;
+
+namespace Microsoft.AspNetCore.Components.Server.JsonSourceGeneration
+{
+    internal partial class JsonContext : JsonSerializerContext, WebEventData.IWebEventJsonSerializerContext
```

Comment:
Is this just to make sure we don't pass the wrong context type (since they all have the same name and are hard to distinguish in source code), or for a more complex reason?

Reply:
You get slightly better typing with this approach. In the absence of this, you have to call `JsonSerializer.Serialize(string, Type, JsonContext)` which would result in a runtime error if the `JsonContext` does not know of the type. 

With this contract, the shared contract is in defined in WebEventData.cs and lets you do `JsonSerializer.Serialize<MyType>(string, JsonPropertyInfo<MyType>)` which feels a bit nicer.